arch/xtensa/src/common/xtensa_dumpstate.c

```diff
@@ -395,6 +395,15 @@ void xtensa_dumpstate(void)
     }
   else
     {
+#ifdef CONFIG_STACK_COLORATION
+      uint32_t remain;
+
+      remain = ustacksize - xtensa_stack_check((uintptr_t)ustackbase,
+                                               ustacksize);
+      ustackbase += remain;
+      ustacksize -= remain;
+#endif
+
       _alert("ERROR: Stack pointer is not within allocated stack\n");
       xtensa_stackdump(ustackbase, ustackbase + ustacksize);
```

Comment:
should we add force to xtensa_stackdump and remove the code dup?

Reply:
Done